### PR TITLE
Updated casaulity - variability - inital combinations to FMI3

### PR DESCRIPTION
**Pull request by PeterMeisrimelModelon**

### CODE REVIEW

File: Test/FMI3/fmi3_causality_variability_initial_test.cpp [hunk 1]
```diff
@@ -164,14 +171,91 @@ TEST_CASE("Default valid combinations for causality = independent") {
...
+    int expected[N_VARIABILITIES] = {0, 1, 1, 0, 0, 0};
+    validate_variability_causality(fmi3_causality_enu_structural_parameter, expected);
 }
 
 TEST_CASE("Default valid combinations for causality = unknown") {
     int expected[N_VARIABILITIES] = {0, 0, 0, 0, 0, 0};
     validate_variability_causality(fmi3_causality_enu_unknown, expected);
 }
 
+#define N_VALIDCVI (12)
```
Comment (PeterMeisrimelModelon) [inline Test/FMI3/fmi3_causality_variability_initial_test.cpp:183]: Best read this new section function by function from the bottom.

File: src/Util/src/FMI3/fmi3_enums.c [hunk 2]
```diff
@@ -148,40 +148,43 @@ const char* fmi3_initial_to_string(fmi3_initial_enu_t ini) {
...
+    /* continuous */{fmi3_initial_enu_unknown, fmi3_initial_enu_unknown,    fmi3_initial_enu_exact  , fmi3_initial_enu_calculated, fmi3_initial_enu_calculated, fmi3_initial_enu_unknown, fmi3_initial_enu_unknown}
 };
 
 FMILIB_EXPORT fmi3_variability_enu_t fmi3_get_default_valid_variability(fmi3_causality_enu_t c)
 {
-    if (c == fmi3_causality_enu_calculated_parameter || c == fmi3_causality_enu_parameter) {
-        return fmi3_variability_enu_fixed;
-    } else {
-        return fmi3_variability_enu_continuous;
+    switch(c){
```
Comment (PeterMeisrimelModelon) [inline src/Util/src/FMI3/fmi3_enums.c:161]: opted for a straight forward switch case, in particular to also address the causality = unknown case.

**Review by filip-stenstrom — CHANGES_REQUESTED** [review]
Good that you test the whole table

File: src/Util/src/FMI3/fmi3_enums.c [hunk 3]
```diff
@@ -148,64 +148,69 @@ const char* fmi3_initial_to_string(fmi3_initial_enu_t ini) {
...
+    if (c >= fmi3_causality_enu_unknown) {
+        return fmi3_variability_enu_unknown;
+    }
+    // Spec: "The default for variables of causality `parameter`, `structural parameter` or `calculated parameter` is `fixed`"
+    if (c <= fmi3_causality_enu_calculated_parameter) {
+        return fmi3_variability_enu_fixed;
+    }
+
+    if (isFloat) {
+        // Spec: "The default for variables of type `<Float32>` and `<Float64>` and causality other than `parameter`, `structural parameter` or `calculated parameter` is <<continuous>>"
```
Comment (filip-stenstrom) [inline src/Util/src/FMI3/fmi3_enums.c:170]: Long line. Tip: Add this to your personal settings.json: `    "editor.rulers": [ 120 ],`

File: src/XML/src/FMI3/fmi3_xml_variable.c [hunk 4]
```diff
@@ -1165,7 +1165,8 @@ static int fmi3_xml_variable_process_attr_causality_variability_initial(fmi3_xml
     }
     if (!fmi3_is_valid_variability_causality(variability, causality)) {
         fmi3_variability_enu_t bad_variability = variability;
-        variability = fmi3_get_default_valid_variability(causality);
+        bool isFloat = (elm_id == fmi3_xml_elmID_Float32Variable) || (elm_id == fmi3_xml_elmID_Float64Variable);
```
Comment (filip-stenstrom) [inline src/XML/src/FMI3/fmi3_xml_variable.c:1168]: Could you use `variable->baseType` instead? I'm trying to reduce places where elmID is used for anything but choosing the element handler + error reporting.

Like this:

`isFloat = (fmi3_base_type_enu_is_float(fmi3_xml_get_variable_base_type(variable)))`

File: Test/FMI3/fmi3_variability_causality_initial_test.cpp [hunk 5]
```diff
@@ -218,59 +233,92 @@ static bool non_default_valid_combination(fmi3_variability_enu_t v, fmi3_causali
...
-    const char* errMsg = fmi3_import_get_last_error(fmu);
-    REQUIRE(strcmp(errMsg, expectedErr) == 0);
+    const char* expectedError;
+    fmi3_import_variable_t* v;
 
-    // check variablity has been set correctly
-    fmi3_import_variable_t* v = fmi3_import_get_variable_by_vr(fmu, 0);
+    // check Float32 type variables
+    expectedError = "Invalid combination of variability 'continuous' and "
+    "causality 'parameter' for variable 'f32_parameter'. Setting variability to 'fixed'";
```
Comment (filip-stenstrom) [inline Test/FMI3/fmi3_variability_causality_initial_test.cpp:286]:
```suggestion
    expectedError = "Invalid combination of variability 'continuous' and "
                               "causality 'parameter' for variable 'f32_parameter'. Setting variability to 'fixed'";
```
or double indent

**Review by filip-stenstrom — APPROVED** [review]

File: src/Util/src/FMI3/fmi3_enums.c (hunk 3 above)
Comment (PeterMeisrimelModelon) [reply src/Util/src/FMI3/fmi3_enums.c]: Fixed.

File: Test/FMI3/fmi3_variability_causality_initial_test.cpp (hunk 5 above)
Comment (PeterMeisrimelModelon) [reply Test/FMI3/fmi3_variability_causality_initial_test.cpp]: Fixed to double indent.

File: src/XML/src/FMI3/fmi3_xml_variable.c (hunk 4 above)
Comment (PeterMeisrimelModelon) [reply src/XML/src/FMI3/fmi3_xml_variable.c:1168]: Using `fmi3_xml_get_variable_base_type` was also my first choice. 
Unfortunately, `variable->type` is only set later on. Not overly happy with this solution, but anything else would require a larger restructuring at this point.

**Pull request merged**